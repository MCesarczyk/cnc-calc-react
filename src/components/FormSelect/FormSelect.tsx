import { Label } from 'ui/atoms/label';
import { TextSub } from 'ui/atoms/textSub';
import { FieldWrapper } from 'ui/atoms/fieldWrapper';
import { Select } from 'ui/atoms/select';

interface FormSelectProps {
  name: string;
  sub?: string;
  unit?: string;
  id: string;
  value: string;
  data: any[];
  onChange: any;
}

export const FormSelect = ({ name, sub, unit, id, value, data, onChange }: FormSelectProps) => {
  return (
    <FieldWrapper>
      <Label>
        <label htmlFor={id}>
          {name}
          <TextSub>{sub}</TextSub>
          {unit}
        </label>
      </Label>
      <Select id={id} value={value} onChange={onChange}>
        {data.map((item) => (
          <option key={item.key}>{item.value}</option>
        ))}
      </Select>
    </FieldWrapper>
  );
};
