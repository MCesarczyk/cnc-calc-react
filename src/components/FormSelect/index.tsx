import { FieldWrapper } from "../FieldWrapper";
import { Label } from "../Label";
import { TextSub } from "../TextSub";
import { Select } from "../Select";

interface FormSelectProps {
  name: string;
  sub?: string;
  unit?: string;
  id: string;
  value: string;
  data: any[];
  onChange: any;
}

const FormSelect = ({
  name,
  sub,
  unit,
  id,
  value,
  data,
  onChange,
}: FormSelectProps) => {
  return (
    <FieldWrapper>
      <Label>
        <label htmlFor={id}>
          {name}
          <TextSub>{sub}</TextSub>
          {unit}
        </label>
      </Label>
      <Select id={id} value={value} onChange={onChange}>
        {data.map((item) => (
          <option key={item.key}>{item.value}</option>
        ))}
      </Select>
    </FieldWrapper>
  );
};

export default FormSelect;
