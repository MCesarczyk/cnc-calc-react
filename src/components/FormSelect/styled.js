import styled from "styled-components";

export const FieldWrapper = styled.label`
    display: flex;
    gap: 5px;
    align-items: center;
    justify-content: space-between;

    @media (max-width: 480px) {
        flex-direction: column;
        justify-content: center;
    }
`;

export const InputLabelText = styled.span`
    text-align: right;

    @media (max-width: 480px) {
        text-align: center;
    }
`;

export const Select = styled.select`
    min-width: 222px;
    border: solid 1px ${({ theme }) => theme.color.primary};
    border-radius: 5px;
    padding: 5px;
`;