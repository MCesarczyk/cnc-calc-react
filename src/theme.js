export const theme = {
    color: {
        primary: "#1e3246",
        background: "#133246",
    },
    breakpoint: {
        mobileMax: "480px",
    },
}