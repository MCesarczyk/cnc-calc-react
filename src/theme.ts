export const theme = {
  color: {
    primary: '#1e3246',
    primaryLight: '#2e4c6b',
    primaryText: '#000',
    contrastText: '#fff',
    background: '#133246',
    elementBackground: '#fff',
    submitButtonText: '#19aa6e',
    resetButtonText: '#dc3223',
    copyButtonText: '#dc143c',
    activeNavlinkText: '#2188FF',
    invalid: 'beige',
    invalidBackground: 'beige',
    disabled: '#ddd',
    disabledText: '#808080',
    disabledBackground: '#ddd',
    placeholder: '#747474',
  },
  breakpoint: {
    xs: '480px',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1400px',
  },
};
