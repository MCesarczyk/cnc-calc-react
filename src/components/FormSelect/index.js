import { TextSub } from "../TextSub";
import { InputLabelText, FieldWrapper, Select } from "./styled";

const FormSelect = ({
  name,
  sub,
  unit,
  id,
  value,
  data,
  onChange
}) => {
  return (
    <FieldWrapper>
      <InputLabelText>
        <label htmlFor={id}>
          {name}
          <TextSub>
            {sub}
          </TextSub>
          {unit}
        </label>
      </InputLabelText>
      <Select
        id={id}
        value={value}
        onChange={onChange}
      >
        {
          data.map(item => (
            <option key={item.key}>
              {item.value}
            </option>
          ))
        }
      </Select>
    </FieldWrapper>
  )
};

export default FormSelect;