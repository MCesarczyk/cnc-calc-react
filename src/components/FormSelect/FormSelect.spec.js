import { fireEvent, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { theme } from 'theme';
import FormSelect from '.';

const data = [
  { key: 0, value: '' },
  { key: 1, value: 'A' },
  { key: 2, value: 'B' },
  { key: 3, value: 'C' },
  { key: 4, value: 'D' },
  { key: 5, value: 'E' },
  { key: 6, value: 'F' },
  { key: 7, value: 'G' },
  { key: 8, value: 'H' },
  { key: 9, value: 'I' },
];

const setup = () =>
  render(
    <ThemeProvider theme={theme}>
      <FormSelect data={data} />
    </ThemeProvider>,
  );

test('Select component should be empty initially', () => {
  const select = setup().getByRole('combobox');
  expect(select.value).toBe('');
});

test('Select component should display selected value', () => {
  const select = setup().getByRole('combobox');
  fireEvent.change(select, { target: { value: 'B' } });
  expect(select.value).toBe('B');

  fireEvent.change(select, { target: { value: 'B' } });
  expect(select.value).toBe('B');

  fireEvent.change(select, { target: { value: 'F' } });
  expect(select.value).toBe('F');
});

test('FormSelect component rendering properly', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <FormSelect data={data} />
    </ThemeProvider>,
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
