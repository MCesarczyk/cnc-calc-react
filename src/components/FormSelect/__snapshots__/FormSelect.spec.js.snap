// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelect component rendering properly 1`] = `
.c0 {
  display: flex;
  gap: 0.5rem;
  align-items: center;
  justify-content: space-between;
}

.c1 {
  text-align: right;
}

.c2 {
  vertical-align: sub;
  font-size: 0.7em;
  margin-right: 0.15em;
}

.c3 {
  min-width: 16rem;
  border: solid 1px #1e3246;
  border-radius: 5px;
  padding: 5px;
}

@media (max-width: 480px) {
  .c0 {
    flex-direction: column;
    justify-content: center;
    gap: 0.25rem;
  }
}

@media (max-width: 480px) {
  .c1 {
    align-self: flex-start;
  }
}

@media (max-width: 480px) {
  .c3 {
    width: 100%;
  }
}

<div
  className="c0"
>
  <label
    className="c1"
  >
    <label>
      <span
        className="c2"
      />
    </label>
  </label>
  <select
    className="c3"
  >
    <option />
    <option>
      A
    </option>
    <option>
      B
    </option>
    <option>
      C
    </option>
    <option>
      D
    </option>
    <option>
      E
    </option>
    <option>
      F
    </option>
    <option>
      G
    </option>
    <option>
      H
    </option>
    <option>
      I
    </option>
  </select>
</div>
`;
